src/BarChart/BarSeries/BarTargetMarker.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { FC, useMemo } from 'react';
+import { motion } from 'framer-motion';
+import {
+  ChartInternalShallowDataShape,
+  DEFAULT_TRANSITION,
+  Direction
+} from '@/common';
+
+export interface BarTargetMarkerProps {
+  /**
+   * Height of the bar.
+   */
+  height: number;
+
+  /**
+   * Width of the bar.
+   */
+  width: number;
+
+  /**
+   * SVG x attribute for the bar.
+   */
+  x: number;
+
+  /**
+   * SVG y attribute for the bar.
+   */
+  y: number;
+
+  /**
+   * Group index or index of the bar. Set internally by `BarSeries`.
+   */
+  index: number;
+
+  /**
+   * D3 scale for Axis. Set internally by `BarChart`.
+   */
+  scale: any;
+
+  /**
+   * Parsed data shape. Set internally by `BarChart`.
+   */
+  data: ChartInternalShallowDataShape;
+
+  /**
+   * Color of the target marker line.
+   */
+  fill: string;
+
+  /**
+   * Color of the positive delta line.
+   */
+  positiveDeltaFill: string;
+
+  /**
+   * Color of the negative delta line.
+   */
+  negativeDeltaFill: string;
+
+  /**
+   * Number of the bars in the bar group. Set internally by `BarSeries`.
+   */
+  barCount: number;
+
+  /**
+   * Layout of bar chart to render. Set internally by `BarSeries`.
+   */
+  layout: Direction;
+
+  /**
+   * Whether to animate the enter/update/exit. Set internally by `BarSeries`.
+   */
+  animated: boolean;
+
+  /**
+   * Class name to apply to the text.
+   */
+  className?: string;
+
+  /**
+   * Thickness of the target marker line.
+   */
+  targetLineThickness?: number;
+
+  /**
+   * Thickness of the difference/delta line that shows distance between target and actual value.
+   */
+  deltaLineThickness?: number;
+}
+
+export const BarTargetMarker: FC<Partial<BarTargetMarkerProps>> = ({
+  data,
+  height,
+  width,
+  x,
+  y,
+  animated,
+  layout = 'vertical',
+  scale,
+  index,
+  barCount,
+  fill = '#fff',
+  positiveDeltaFill = '#00C49F',
+  negativeDeltaFill = '#FF7361',
+  targetLineThickness = 2,
+  deltaLineThickness = 1
+}) => {
+  const isVertical = layout === 'vertical';
+  const valuePosition = useMemo(() => scale(data.value), [data.value, scale]);
```

Comment:
Might be worth combining these memos since they will affect the same prop. 

```
const [valuePos, targetPos] = useMemo(() => [scale(data.value),scale(data.target), ], [whatever]);
```